Cvar for automatically untoggling inputs on teleporting via a Momentum teleport

braem opened the pull request:
Closes #922

Only untoggles if the teleport is a `CTriggerMomentumTeleport` as not all teleporting of the player should reset inputs.

Uses defined but unused `AfterTeleport` function.

Probably should've been a good first issue but I had already assigned to myself.

### Checklist
- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

Gocnak commented:
I think this should wait until tricksurf gets merged in as there's also a new `ManualTeleport` function that hooks into things like savelocs and mom_restart. In the meantime you can move the "ResetToggledKeys" logic to the player.

braem commented:
> 
> 
> I think this should wait until tricksurf gets merged in as there's also a new `ManualTeleport` function that hooks into things like savelocs and mom_restart. In the meantime you can move the "ResetToggledKeys" logic to the player.

Oo that sounds nifty. Not sure what you mean to put in player here, just a method that checks the cvar then resets toggles?

braem commented:
Dropped the commit and pushed 2 new ones:

- add bool field to teleport that indicates whether the tele is for failing a level, default off.
- added toggled buttons to saveloc state

I will convert this PR to draft for now until the `ManualTeleport` stuff is in.

Gocnak commented:
This should probably also apply when the user hits a `trigger_momentum_teleport_progress` as well, as those are inherently reset triggers.

braem commented:
Now resets player's toggled inputs when teleported by a fail teleport. Didn't use a cvar as I figure the "fail" bool suffices.
